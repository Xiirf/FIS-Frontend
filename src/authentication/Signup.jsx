import React from 'react';

function Signup() {
  return (
    <div>
      <form>
        <input placeholder="login" />
        <input type="email" placeholder="email" />
        <input type="password" placeholder="password" />
        <input type="password" placeholder="password again" />
        <button type="submit">Sign Up</button>
      </form>
      <a href="/signin">Already have an account?</a>
    </div>
  );
}

export default Signup;
