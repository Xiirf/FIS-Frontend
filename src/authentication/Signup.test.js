import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import { act } from "react-dom/test-utils";
import Signup from './Signup';
import { create } from "react-test-renderer";
import { render, fireEvent, wait } from '@testing-library/react';

let container = null;
beforeEach(() => {
  // met en place un élément DOM comme cible de rendu
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // nettoie en sortie de test
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('renders without crashing', () => {
  render(<Signup />, container);
});

it("matches the snapshot", () => {
    const signup = create(<Signup />);
    expect(signup.toJSON()).toMatchSnapshot();
});
/*
it("submits correct values", async () => {
    const mockFunction = jest.fn(v => {
        console.log('this prints later')
        // done()
    })

    const { container } = render(<Signup />);

    console.log(container)

    const name = container.querySelector('input[name="username"]')
    const email = container.querySelector('input[name="email"]')
    const password = container.querySelector('input[name="password"]')
    const submit = container.querySelector('button[type="submit"]')
    const results = container.querySelector("textarea");

    console.log(name)
  
    await act(() => {
      fireEvent.change(name, {
        target: {
          value: "mockname"
        }
      })
    })
  
    await act(() => {
      fireEvent.change(email, {
        target: {
          value: "mock@email.com"
        }
      })
    })
  
    await act(() => {
      fireEvent.change(password, {
        target: {
          value: "mockpassword"
        }
      })
    })
  
    await act(() => {
      fireEvent.click(submit)
    })
  
    expect(results.innerHTML).toBe(
      '{"username":"mockname","password":"mockpassword","email":"mock@email.com"}'
    )
  })*/