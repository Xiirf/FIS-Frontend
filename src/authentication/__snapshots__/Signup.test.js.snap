// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div>
  <h2>
    Inscribirse
  </h2>
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="username"
      >
        Nombre de usuario
      </label>
      <input
        className="form-control"
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="password"
      >
        Contraseña
      </label>
      <input
        className="form-control"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Correo electrónico
      </label>
      <input
        className="form-control"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        Crear una cuenta
      </button>
    </div>
  </form>
</div>
`;
